src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -1088,12 +1092,16 @@ public async Task ReceiveIovMaxUdp_SuccessOrMessageSize()
             using (sender)
             {
                 sender.Connect(receiverEndPoint);
+
+                // Synchronize and wait for receiving task to be ready.
+                b.SignalAndWait();
+
                 var sendBuffer = new byte[SegmentCount];
                 for (int i = 0; i < 10; i++) // UDPRedundancy
                 {
                     int bytesSent = sender.Send(sendBuffer);
                     Assert.Equal(SegmentCount, bytesSent);
-                    await Task.WhenAny(receiveTask, Task.Delay(1));
+                    await Task.WhenAny(receiveTask, Task.Delay(3));
```

Comment:
Note that I mislead you earlier today: I accidentally said this would wait for a second, but this argument is in milliseconds, so this is only increasing it from 1ms to 3ms.  If there's a concern here about how long things take in an overloaded CI system, you might want to make this significantly larger.